### PR TITLE
front: rollingstockeditor: isolate search filters from rollingstockmodal to use it in rollingstockeditor

### LINKED ISSUE #4462: isolate rollingstockmodal filter to use it in rollingstockeditor
we can't actually filter the rollingstock list in rollingstockeditor.

isolating the filter functions in rollingstockmodal could let use use it in rollingstockeditor.

### PR DESCRIPTION
**Pull request by Yohh**
close #4462

### CODE REVIEW

File: front/src/common/RollingStockSelector/SearchRollingStock.tsx [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+  const searchMateriel = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setFilters({ ...filters, text: e.target.value.toLowerCase() });
+    if (setIsLoading) setIsLoading(true);
+  };
+  const toggleFilter = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setFilters({ ...filters, [e.target.name]: !filters[e.target.name as 'elec' | 'thermal'] });
+    if (setIsLoading) setIsLoading(true);
+  };
+
+  function rollingStockPassesSearchedStringFilter(
```
Comment (anisometropie) [inline front/src/common/RollingStockSelector/SearchRollingStock.tsx]: keep `rollingStockPassesSearchedStringFilter`, `filterRollingStocks`, `rollingStockPassesEnergeticModeFilters` outside of the component, and bunched together as they are related

File: front/src/common/RollingStockSelector/SearchRollingStock.tsx [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+      metadata?.series,
+      metadata?.type,
+      metadata?.grouping,
+    ].some(includesSearchedString);
+  }
+
+  function filterRollingStocks(rollingStockList: LightRollingStock[], filterList: Filters) {
+    return rollingStockList?.filter(({ name, metadata, effort_curves: effortCurves }) => {
+      const passSearchedStringFilter = rollingStockPassesSearchedStringFilter(
+        name as string,
```
Comment (anisometropie) [inline front/src/common/RollingStockSelector/SearchRollingStock.tsx]: already a string

**Review by anisometropie — CHANGES_REQUESTED** [review]
Really glad to see this refacto done, and done well

File: front/src/common/RollingStockSelector/SearchRollingStock.tsx (hunk 1 above)
Comment (Yohh) [reply front/src/common/RollingStockSelector/SearchRollingStock.tsx]: done

File: front/src/common/RollingStockSelector/SearchRollingStock.tsx (hunk 2 above)
Comment (Yohh) [reply front/src/common/RollingStockSelector/SearchRollingStock.tsx]: done

**Review by anisometropie — APPROVED** [review]

**Pull request merged**